src/libs/portfolio/pendingAmountsHelper.ts

```diff
@@ -0,0 +1,89 @@
+export type PendingAmounts = {
+  isPending: boolean
+  pendingBalance: bigint
+  pendingToBeSigned?: bigint
+  pendingToBeConfirmed?: bigint
+}
+
+/**
+ * Function for calculating the pending balance and the delta amounts
+ * for pendingToBeSigned and pendingToBeConfirmed states (referred to as token badges) of a token.
+ *
+ * While calculating the balance is a routine operation,
+ * calculating the pending delta and determining its state (whether it needs to be signed or confirmed)
+ * can be quite challenging.
+ *
+ * We use this information to display PendingToBeSigned, PendingToBeConfirmed, or both badges on the Dashboard.
+ *
+ * Let's review the main scenarios where we encounter delta amounts, followed by a discussion of corner cases.
+ *
+ * Main scenarios:
+ * 1. If there is an AccOp that has not yet been signed, we return the `pendingToBeSigned` amount.
+ * 2. If there is an AccOp that has been signed and broadcasted (but not yet confirmed),
+ *    we display the `pendingToBeConfirmed` badge, reflecting the corresponding amount.
+ * 3. If we detect a delta between the pending and latest token amounts, it indicates that there is something awaiting confirmation.
+ *    In this case, we also return the `pendingToBeConfirmed` amount.
+ *    For example, if someone sends you tokens outside of the extension, the extension will show the `pendingToBeConfirmed` amount.
+ *
+ * Rare scenarios:
+ * 1. It is possible to have both `pendingToBeSigned` and `pendingToBeConfirmed` badges in the following scenario:
+ *    - If someone sends you tokens, the `latestPendingDelta` is triggered.
+ *      At the same time, if you have an AccOp waiting to be signed, both badges will be displayed,
+ *       and the amounts for `pendingToBeSigned` and `pendingToBeConfirmed` will be calculated.
+ * 2. If you sign and broadcast an AccOp while the `latestPendingDelta` also has a value,
+ *    the `pendingToBeConfirmed` badge will be displayed, representing the sum of both `simulationDelta` and `pendingLatestDelta`.
+ *    - This is similar to the previous case, but here the AccOp has been broadcasted.
+ *      When there is a delta between the latest and pending block amounts, both deltas are summed, resulting in the `pendingToBeConfirmed` amount.
+ */
```

Comment:
This documentation is really helpful but hard to digest because of the amount of information. I know there are a lot of cases but imo it'd be much better if we could move the comments next to the code that is being explained.